### PR TITLE
Add MinVersion field to muxer configs.

### PR DESCRIPTION
**Pull request by john-floren-gravwell**
Muxer will refuse to connect to any indexer reporting an API version lower than that specified in the config, thus if you intend to use ditto you might say:

		igCfg := ingest.UniformMuxerConfig{
			[...]
			MinVersion:         ingest.MINIMUM_DITTO_VERSION,
			[...]
		}
		igst, _ = ingest.NewUniformMuxer(igCfg)

An insufficient API version is treated as non-fatal; the muxer will attempt to reconnect to that indexer periodically, on the assumption that it will eventually be updated.

As long as one indexer meets the minimum version requirement, you will be able to use the muxer as usual. If not, the muxer will never go hot.

### CODE REVIEW

**Comment by john-floren-gravwell** [comment]
Addresses #1442

**Review by david-fritz-gravwell — APPROVED** [review]
LGTM

File: ingest/muxer.go [hunk 1]
```diff
@@ -1831,6 +1836,24 @@ loop:
 			}
 			continue
 		}
+		// Make sure the version is new enough
+		if ig.ew.serverVersion < im.minVersion {
```
Comment (kris-watts-gravwell) [inline ingest/muxer.go:1840]: I think this needs an additional check.  

If minVersion > 0 && serverVersion < minVersion

**Review by kris-watts-gravwell — CHANGES_REQUESTED** [review]
It looks like this will always fail if min version is not set.   I would argue we only do that check if minVersion is set

**Review by kris-watts-gravwell — APPROVED** [review]
LGTM

File: ingest/muxer.go (hunk 1 above)
Comment (kris-watts-gravwell) [reply ingest/muxer.go:1840]: yes I am a moron and yes I realized it several hours later.

**Pull request merged**